src/commands/dev/edge/watch.rs

```diff
@@ -38,15 +38,26 @@ pub fn watch_for_changes(
         // this allows the server to route subsequent requests
         // to the proper script
         let uploaded = setup::upload(&mut target, &deploy_target, &user, session_token, verbose);
+
         match uploaded {
             Ok(token) => {
                 *preview_token = token;
             }
             Err(err) => {
                 if let Some(err) = err.downcast_ref::<setup::BadRequestError>() {
-                    // no need to refresh session here, bad request indicates problem with user code
-                    StdOut::warn(&err.0);
+                    // if the API error code is 10049, then it's an expired preview token
+                    if err.0.contains("10049") {
+                        refresh_session_channel.send(Some(()))?;
+                        break;
+                    }
+
+                    // otherwise it is a non recoverable error
+                    StdOut::warn(&format!(
+                        "{}\nPlease terminate `wrangler dev` using Ctrl+C.",
+                        &err.0
```

Comment:
Shouldn't this be a hard error?
```suggestion
                    // otherwise it is a non recoverable error
                    anyhow::bail!(err.0);
```

Reply:
Not sure what all the possible API errors are, but we can add a case for wrangler to still continue watching if there's a syntax error.

Reply:
I tried `anyhow::bail!(err.0);`, but it will not shut down `wrangler dev` as it will just return from the spawned thread.